dbms/src/Interpreters/PredicateExpressionsOptimizer.cpp

```diff
@@ -146,9 +146,13 @@ bool PredicateExpressionsOptimizer::allowPushDown(
         return false;
     else
     {
-        ASTPtr expr_list = ast_select->select();
         ExtractFunctionVisitor::Data extract_data;
-        ExtractFunctionVisitor(extract_data).visit(expr_list);
+
+        ASTPtr expression_list = std::make_shared<ASTExpressionList>();
+        expression_list->children.push_back(subquery->select());
+        expression_list->children.push_back(ast_select->select());
```

Comment:
Do I understand correctly that pushdown is not applicable if there is a stateful function in external or internal SELECT clauses? Ok, but it needs to be commented in code.

And what about WHERE clause? For example, inner subquery can filter data based on `neighbour` function and we pushed down additional condition in WHERE clause of subquery and while evaluating subquery, this condition was moved to PREWHERE clobbering the row numbers of `neighbour` function.

Reply:
I'll close it, it involves some other modify. I'm not ready yet.